src/server/utils/devCdnFactory.js

```diff
@@ -0,0 +1,199 @@
+    module.node.url = module.node.url.replace('[one-app-dev-cdn-url]', oneAppDevCdnAddress);
+  });
+  return JSON.stringify(moduleMap, null, 2);
+};
+
+const matchPathToKnownRemoteModuleUrl = (
+  incomingRequestPath,
+  remoteModuleBaseUrls
+) => remoteModuleBaseUrls.find((remoteModuleBaseUrl) => {
+  const remoteModuleUrlOrigin = new URL(remoteModuleBaseUrl).origin;
+  const remoteModulePath = remoteModuleBaseUrl.replace(remoteModuleUrlOrigin, '');
+
+  return incomingRequestPath.startsWith(remoteModulePath);
+});
+
+const consumeRemoteRequest = async (remoteModuleMapUrl, hostAddress, remoteModuleBaseUrls) => {
+  try {
+    const response = await fetch(remoteModuleMapUrl, {
+      agent: new ProxyAgent(),
+    });
+    // clear out remoteModuleBaseUrls as the new module map now has different urls in it
+    // not clearing would result in an ever growing array
+    remoteModuleBaseUrls.splice(0, remoteModuleBaseUrls.length);
+    const remoteModuleMap = JSON.parse(response.body);
+    const { modules } = remoteModuleMap;
+    const oneAppDevStaticsAddress = `${hostAddress}/static`;
+    Object.keys(modules).map((moduleName) => {
+      const module = modules[moduleName];
+
+      const parsedBundlePath = path.parse(module.node.url, '');
+
+      // store urls for later lookup when module requests are caught by one-app-dev-cdn
+      remoteModuleBaseUrls.push(
+        module.node.url.replace(parsedBundlePath.base, '')
+      );
+
+      // override remote module map to point all module URLs to one-app-dev-cdn
+      module.node.url = module.node.url.replace(
+        new URL(module.node.url).origin, oneAppDevStaticsAddress
+      );
+      module.legacyBrowser.url = module.legacyBrowser.url.replace(
+        new URL(module.legacyBrowser.url).origin, oneAppDevStaticsAddress
+      );
+      module.browser.url = module.browser.url.replace(
+        new URL(module.browser.url).origin, oneAppDevStaticsAddress
+      );
+      return module;
+    });
+    return remoteModuleMap;
+  } catch (error) {
+    console.warn(
+      `one-app-dev-cdn error loading module map from ${remoteModuleMapUrl}: ${error}`
+    );
+    return {};
+  }
+};
+
+export const oneAppDevCdnFactory = ({
+  localDevPublicPath,
+  remoteModuleMapUrl,
+  useLocalModules,
+  appPort,
+  useHost,
+  routePrefix = '',
+}) => {
+  if (!remoteModuleMapUrl && !useLocalModules) {
+    throw new Error('remoteModuleMapUrl is a required param when useLocalModules is not true');
+  }
+  if (!localDevPublicPath) { throw new Error('localDevPublicPath is a required param'); }
+  if (!appPort) { throw new Error('appPort is a required param'); }
+
+  if (remoteModuleMapUrl) {
+    console.log(`one-app-dev-cdn loading module map from ${remoteModuleMapUrl}`);
+  } else {
+    console.log('one-app-dev-cdn only using locally served modules');
+  }
+
+  const oneAppDevCdn = Fastify();
+
+  if (process.env.NODE_ENV === 'production') {
```

Comment:
this check is already in place (see https://github.com/americanexpress/one-app/blob/feature/express-fastify-migration/src/server/index.js#L149). I don't think this is very useful since it returns an empty fastify instance, the server can be initialized but no routes will be available. I suggest to just throw an error, and even before creating the Fastify instance